system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp

```diff
@@ -185,12 +184,12 @@ void MrmHandler::operateMrm()
     if (current_mrm_behavior == mrm_state_.behavior) {
       return;
     }
-    if (!requestMrmBehavior(mrm_state_.behavior, RequestType::CANCEL)) {
+    if (
```

Comment:
The following is more straghtforward:

```
if (requestMrmBehavior(current_mrm_behavior, RequestType::CALL)) {
  mrm_state_.behavior = current_mrm_behavior;
} else {
  handleFailedRequest();
}
```

Reply:
It has been fixed. Thank you.

Reply:
Sorry but my suggestion will change the behavior, since `requestMrmBehavior` has side effects.
Please revert it.

Reply:
Fix it as the following

```
    // First: cancel the previous behavior
    if (!requestMrmBehavior(mrm_state_.behavior, RequestType::CANCEL)) {
      handleFailedRequest();
      return;
    }

    // Second: call the current behavior
    if (!requestMrmBehavior(current_mrm_behavior, RequestType::CALL)) {
      handleFailedRequest();
      return;
    }

    mrm_state_.behavior = current_mrm_behavior;
    return;
```

Reply:
Thank you for the verification. I have corrected the above.